lib/utils/format.ts

```diff
@@ -0,0 +1,256 @@
+
+    // note on 'any': TypeScript < 4.4 does not support using Symbol for keys.
+    // Cast can changed back to 'symbol | string' in sept. 2022 when we drop support for < 4.4
+    const columnNameOrOperator: any = rawAttribute?.field ?? attributeNameOrOperator;
+
+    if (
+      isPlainObject(where[attributeNameOrOperator])
+        && !(
+          rawAttribute
+          && (rawAttribute.type instanceof DataTypes.HSTORE
+            || rawAttribute.type instanceof DataTypes.JSON)
+        )
+    ) {
+      // Prevent renaming of HSTORE & JSON fields
+      newAttributes[columnNameOrOperator] = mapOptionFieldNames(
+        {
+          where: where[attributeNameOrOperator],
+        },
+        Model,
+      ).where;
+
+      continue;
+    }
+
+    if (Array.isArray(where[attributeNameOrOperator])) {
+      newAttributes[attributeNameOrOperator] = [...where[attributeNameOrOperator]];
+
+      for (const [index, wherePart] of where[attributeNameOrOperator].entries()) {
+        if (isPlainObject(wherePart)) {
+          newAttributes[columnNameOrOperator][index] = mapWhereFieldNames(wherePart, Model);
+        }
+      }
+
+      continue;
+    }
+
+    newAttributes[columnNameOrOperator] = where[attributeNameOrOperator];
+  }
+
+  return newAttributes;
+}
+
+/**
+ * getComplexKeys
+ *
+ * @param  {object} obj
+ * @returns {Array<string|symbol>} All keys including operators
+ * @private
+ */
+export function getComplexKeys(obj: object): Array<string | symbol> {
+  return [
+    ...getOperators(obj),
+    ...Object.keys(obj),
+  ];
+}
+
+/**
+ * getOperators
+ *
+ * @param  {object} obj
+ * @returns {Array<symbol|string>} All operators properties of obj
+ * @private
+ */
+export function getOperators(obj: object): symbol[] {
+  return Object.getOwnPropertySymbols(obj).filter(s => operatorsSet.has(s));
+}
+
+export function combineTableNames(tableName1: string, tableName2: string): string {
+  return tableName1.toLowerCase() < tableName2.toLowerCase()
+    ? tableName1 + tableName2
+    : tableName2 + tableName1;
+}
+
+/**
+ * Used to map field names in values
+ *
+ * @param dataValues
+ * @param attributeNames
+ * @param ModelClass
+ */
+export function mapValueFieldNames( // TODO: rename to mapAttributesToColumNames?
+  dataValues: Record<string, any>,
+  attributeNames: string[], // TODO: can attributes be symbols too?
+  ModelClass: ModelStatic<Model>,
+): Record<string, any> {
+  const values: Record<string, any> = Object.create(null);
+
+  for (const attributeName of attributeNames) {
+    if (dataValues[attributeName] !== undefined && !ModelClass._virtualAttributes.has(attributeName)) {
+      // Field name mapping
+      const columnName = ModelClass.rawAttributes[attributeName]?.field ?? attributeName;
+
+      values[columnName] = dataValues[attributeName];
+    }
+  }
+
+  return values;
+}
+
+// TODO: rename to removeNullishValues as it also removes undefined.
```

Comment:
Is this something to be addressed in this PR?

Reply:
I think that one I'm just going to rename without external input. It would match the name of the nullish coalescing operator, so there's precedent

Reply:
Yeah, I think the updated name is better